### DISCUSSION TITLE
Is it possible to hide certain properties from autocompletions based on the value of a different property?

### DISCUSSION THREAD

**Discussion by Xterionix** [json-schema-org/community#475] [Q&A] [answered]
```json
{
	"properties": {
		"type": {
			"type": "string"
		},
		"class": {
			"type": "string"
		}
	}
}
```
I want the class property to only show in the autocompletions if type is set to "values:less_than". Is something like that possible, or is that not related to json schemas? For context I'm using VSC

**Comment by gregsdennis** [comment, marked as answer]
You might be able to use `if`/`then`/`else`.

```jsonc
{
  // ...
  "if": {
    "properties": {
      "type": { "not": { "const": "values:less_than" } }
    },
    "required": [ "type" ]
  },
  "then": {
    "properties": {
      "class": false
    }
  }
}
```

That would make validations fail when `type` is not `values:less_than` and `class` is present.  How VSCode responds to such a schema is unknown.  You'll probably need to go to them if it's not supported.

**Reply by Xterionix** [reply]
Thanks
